src/main/webapp/scripts/matrixController.js

```diff
@@ -1,32 +1,179 @@
+var alive = "<td class = \"matrixCell\"><img src = \"../images/alive normal cell.png\" " + cellZoom+"class = \"table-image\" ";
+var death = "<td class = \"matrixCell\"><img src = \"../images/dead cell.png\" "+cellZoom+"class = \"table-image\" ";
+var cellClosure = "></td>";
 var rows = 0;
 var columns = 0;
-var matrix = "";
+var selectedCellType = "deadCell"
+var matrix = [[0,0,0],[0,0,0],[0,0,0]];
+var playSpeed = 1000;
+var cellWithMask = false;
+var cellWithVaccine = false;
+var playing = false;
+var timeController = null;
+var cellDictionary ={
+    0: "../images/dead cell.png",
+    1: "../images/alive normal cell.png",
+    2: "../images/alive mask.png",
+    3: "../images/alive vaccine.png",
+    4: "../images/alive mask vaccine.png",
+    5: "../images/Infected Cell.png",
+    6: "../images/infected mask.png",
+    7: "../images/infected vaccine.png",
+    8: "../images/infected mask vaccine.png"
+};
 
+function generateRandomMatrix(){
+    for(let row = 0; row < rows; row++){
+        for(let column = 0 ; column < columns ; column++ ){
+            document.getElementById(row.toString() + column.toString()).src = cellDictionary[ Math.floor( Math.random() * 8 ) ];
+        }
+    }
+    gameLogic();
+}
 
-function updateMatrix(){
+function gameLogic(){
+    if(playing){
+        timeController = setTimeout(() => {
+                                                if(playing){
+                                                    generateRandomMatrix();
+                                                }
+                                            }, playSpeed);
+    }
+}
+
+function startTheGame(){
+    if(!playing){
+        playing = true;
+        gameLogic();
+    }
+}
+
+function stopTheGame(){
+    playing = false;
+    clearInterval(timeController);
+
+}
+
+function updateCellModifier(target){
+    if(target == "mask")
+        cellWithMask = !cellWithMask;
+    else
+        cellWithVaccine = !cellWithVaccine;
+    console.log(cellWithMask);
+    console.log(cellWithVaccine);
+}
 
+function updateMatrix(){
+    matrix = []
     rows = document.getElementById("matrixRows").value;
     columns = document.getElementById("matrixColumns").value;
-    matrix = "";
+    let matrixToTable = "";
+    let updateFunction = "onclick = \"updateMatrixCell(this)\"";
     for(let row = 0; row < rows; row++){
-        matrix += rowOpening;
+        let tempRow = []
+        matrixToTable += rowOpening;
         for(let column = 0; column < columns; column++){
-            matrix += death;
+            tempRow.push(0)
+            matrixToTable += death + "id = \"" + row.toString() + column.toString() + "\" " + updateFunction + cellClosure;
         }
-        matrix += rowClosure;
+        matrix.push(tempRow);
+        matrixToTable += rowClosure;
     }
-    document.getElementById("matrixTable").innerHTML = matrix;
+    document.getElementById("matrixTable").innerHTML = matrixToTable;
```

Comment:
This is (re)building the entire HTML table in string form every time the table is update. It seems like it should be possible to build the table once, then update the existing table w/ new state. I'd expect that to be faster and I'm pretty sure it'd be easier to understand/maintain. (Otoh, you might get a wave of changes across the UI instead of updating all of the cells in one shot.)